tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/pull/PullService.java

```diff
@@ -0,0 +1,46 @@
+package com.dotcms.api.client.pull;
+
+import com.dotcms.cli.common.OutputOptionMixin;
+import com.dotcms.model.pull.PullOptions;
+import java.io.File;
+import java.util.Map;
+
+public interface PullService {
```

Comment:
why not use the T here instead of a method level?

Reply:
Defining `T` at the interface level would constrain our `PullServiceImpl` to a single content type, thereby losing the generality to handle any type `T`. We would need separate implementations for each content type, which is less scalable and flexible than our current method-level generic approach.